### PR TITLE
Enable starting and stopping ray with "ray start" and "ray stop".

### PR DESCRIPTION
**Pull request by robertnishihara**
This replaces the `start_ray.sh` and `stop_ray.sh` scripts with `ray start` and `ray stop`. These new commands work without cloning the repository.

### CODE REVIEW

File: docker/deploy/Dockerfile [hunk 1]
```diff
@@ -7,4 +7,4 @@ ADD git-rev /ray/git-rev
 WORKDIR /ray/python
 RUN python setup.py install
 WORKDIR /ray
-RUN echo "tail -f /dev/null" >> scripts/start_ray.sh
```
Comment (robertnishihara) [inline docker/deploy/Dockerfile]: @jssmith, can you explain what is going on in this line?

Comment (robertnishihara) [inline docker/deploy/Dockerfile]: Ok, is this correct?
- The command `tail -f /dev/null` just hangs forever.
- So we assume that someone will start the ``ray/deploy` container by running something like
    ```
    docker run ray-project/deploy /ray/scripts/start_ray.sh --head
    ```
    but the problem is that when `start_ray.sh` finishes, the container exits.
- So by appending a command that hangs forever to the end of `start_ray.sh`, we ensure that the container stays alive.

**Comment by raopku** [comment]
how to use ray start and ray stop?

**Comment by robertnishihara** [comment]
It seems like ray is not properly installed. You might try doing

```
pip install ray
```

or

```
cd python
python setup.py install
```

Make sure you uninstall the old version first with `pip uninstall ray`.

If that doesn't fix the problem, please create an issue where we can look into this more!

**Comment by raopku** [comment]
I installed in the second way previously.
```
pip uninstall ray
cd python
python setup.py install --user
```
it seems that ray has been installed successfully except ray start.
```
>>> import ray
>>> ray.init()
```

then i uninstall ray
and reinstall ray using 
```
pip uninstall ray
cd python
python setup.py install
```
ray start command works.

**Comment by robertnishihara** [comment]
Ok great! Perhaps the installation doesn't reliably work when it's already installed.

**Comment by raopku** [comment]
Thank you very much!

**Comment by raopku** [comment]
install ray in the pip way
ray.init() failed.

(1) first we check that ray has been uninstalled.
```
[root@SZV1000274398 ~]# python
Python 2.7.12 |Anaconda 4.2.0 (64-bit)| (default, Jul  2 2016, 17:42:40) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import ray
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named ray
>>> quit()
```

```
(2) install ray using pip
[root@SZV1000274398 ~]# pip install ray
Collecting ray
  Using cached ray-0.1.1-cp27-cp27mu-manylinux1_x86_64.whl
Collecting funcsigs (from ray)
  Downloading funcsigs-1.0.2-py2.py3-none-any.whl
Collecting click (from ray)
  Downloading click-6.7-py2.py3-none-any.whl (71kB)
    100% |████████████████████████████████| 71kB 183kB/s 
Collecting numpy (from ray)
  Downloading numpy-1.12.1-cp27-cp27mu-manylinux1_x86_64.whl (16.5MB)
    100% |████████████████████████████████| 16.5MB 54kB/s 
Requirement already satisfied: psutil in ./anaconda2/lib/python2.7/site-packages (from ray)
Requirement already satisfied: colorama in ./anaconda2/lib/python2.7/site-packages (from ray)
Collecting flatbuffers (from ray)
  Downloading flatbuffers-2015.12.22.1-py2.py3-none-any.whl
Collecting cloudpickle>=0.2.2 (from ray)
  Downloading cloudpickle-0.3.1-py2.py3-none-any.whl
Requirement already satisfied: redis in ./.local/lib/python2.7/site-packages/redis-2.10.5-py2.7.egg (from ray)
Installing collected packages: funcsigs, click, numpy, flatbuffers, cloudpickle, ray
  Found existing installation: cloudpickle 0.2.1
    DEPRECATION: Uninstalling a distutils installed project (cloudpickle) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
    Uninstalling cloudpickle-0.2.1:
      Successfully uninstalled cloudpickle-0.2.1
Successfully installed click-6.7 cloudpickle-0.3.1 flatbuffers-2015.12.22.1 funcsigs-1.0.2 numpy-1.12.1 ray-0.1.1
```

Test ray.
```
[root@SZV1000274398 ~]# python
Python 2.7.12 |Anaconda 4.2.0 (64-bit)| (default, Jul  2 2016, 17:42:40) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import ray
>>> ray.init()
117037:M 06 Jun 17:19:19.743 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
117037:M 06 Jun 17:19:19.743 # Server started, Redis version 3.9.102
117037:M 06 Jun 17:19:19.743 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
Waiting for redis server at 127.0.0.1:39383 to respond...
117041:M 06 Jun 17:19:19.849 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
117041:M 06 Jun 17:19:19.849 # Server started, Redis version 3.9.102
117041:M 06 Jun 17:19:19.849 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
Waiting for redis server at 127.0.0.1:57879 to respond...
Allowing the Plasma store to use up to 3.06033GB of memory.
Connection to socket failed for pathname /tmp/plasma_store85887765
Starting local scheduler with 16 CPUs and 0 GPUs.
Connection to socket failed for pathname /tmp/plasma_store85887765
Connection to socket failed for pathname /tmp/plasma_store85887765
Could not connect to socket /tmp/plasma_store85887765
Could not connect to socket /tmp/plasma_store85887765
Could not connect to socket /tmp/plasma_store85887765
[root@SZV1000274398 ~]# [WARN] (/ray/src/global_scheduler/global_scheduler.cc:414) Missed too many heartbeats from local scheduler, marking as dead.
WARNING:root:Removed local_scheduler, client ID a1babade9ac5de96a23e3d13a405bd665a799d9f

```

ray.init() fails
however, when i install in the `python setup.py install`, ray.init() success.
the os is Ubuntu14.04

Is this my own machine-specific phenomenon?

**Comment by robertnishihara** [comment]
This seems like a bug with Ray, want to create a github issue to track this?

**Comment by atumanov** [comment]
@raopku , I noticed you are installing and testing Ray as root. Could you try doing this as a regular user? (Don't forget to install anaconda for that user and make sure that python and pip binary paths are both pointing to the local user anaconda installation).

**Comment by robertnishihara** [comment]
@raopku, this problem should be fixed now (see #696)!

**Comment by raopku** [comment]
thank you

**Pull request merged**